lib/Onyx.js

```diff
@@ -966,15 +1006,16 @@ function set(key, value) {
     const hasChanged = cache.hasValueChanged(key, valueWithNullRemoved);
 
     // This approach prioritizes fast UI changes without waiting for data to be stored in device storage.
-    broadcastUpdate(key, valueWithNullRemoved, hasChanged, 'set');
+    const updatePromise = broadcastUpdate(key, valueWithNullRemoved, hasChanged, 'set');
 
     // If the value has not changed, calling Storage.setItem() would be redundant and a waste of performance, so return early instead.
     if (!hasChanged) {
-        return Promise.resolve();
+        return updatePromise;
```

Comment:
NAB, I am a _little_ skeptical of this change mainly because it is now harder to reason about when this promise will resolve. 

Before we can easily see it will either resolve right away or after the value is saved to storage. Now it's less clear... it's still both of those, but maybe also however long it takes the queue to flush? Ultimately, I don't know if this concern is valid or if it makes any difference to the existing code we have.

Reply:
we will find out I guess 😄